src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -467,7 +467,7 @@ internal void AddAsset(List<AssetData> assets, string assetName, string version,
             });
         }
 
-        private BuildData MergeBuildManifests(List<BuildData> buildsMetadata)
```

Comment:
We don't need to test all these private methods...  They are private, so they are presumably not part of the contract.  Ideally tests can avoid testing private implementation details when possible, because it makes the source code very hard to maintain... if you can't refactor anything without having to touch a dozen tests, that really makes it hard to keep the code clean and up to date.

In general, we should be testing the contracts of classes, not the implenentations.  These tasks are tough, because we don't really _have_ a contract... it's just "Call execute, and everything happens".

I think they need to be refactored before they are tested, so that the code is structured in more clearly defined chunks of behavior, and then those chunks can be tested.

Testing without doing that refactoring makes the refactoring much harder to do later, meaning it's even less likely to happen, which means better tests are unlikely to get written (because you can't make those better tests without doing the refactor). :-(

Reply:
As you said, testing the public logic with unit tests is basically impossible in this case because it's just a single huge "do all the things" method. If refactoring is what makes the most sense then I'd appreciate some help talking through what that would involve and costing it to see if it fits into the schedule for my v-team.

Reply:
Ick.  This task is especially messy because it's reading magic environment variables rather than accepting task parameters.  Testing this is going to suuuuuck.

Reply:
I'm trying to look at this class... but it's a pretty big class.  It's almost 100 lines long and most of the methods are a bit hard to wrap my head around. :-(

Reply:
I'm there with you - it took me a while to break down and the only reasonable approach I could come up with is to test all the little methods and hope there isn't too much logic left in the Task call when that's done.